### PR TITLE
SelectMultiple - adjust behavior when lazy loading

### PR DESCRIPTION
**Pull request by jcfilben**
#### What does this PR do?
When using the `onMore` prop with SelectMultiple we shouldn't show the 'Select All' button, because without all the options loaded we don't have a good way to select all of them. In addition we shouldn't show the total number of options when lazy loading because the total is unknown.

This PR also adjusts the alignment of the close arrow when using `showSelectedInline`.

#### What testing has been done on this PR?
Added storybook story SelectMultiple/Long list

#### Do Jest tests follow these best practices?

- [ ] `screen` is used for querying.
- [ ] The correct query is used. (Refer to [this list of queries](https://testing-library.com/docs/queries/about/#priority))
- [ ] `userEvent` is used in place of `fireEvent`.
- [ ] `asFragment()` is used for snapshot testing.

#### Do the grommet docs need to be updated?
no
#### Should this PR be mentioned in the release notes?
no, component hasn't been released yet
#### Is this change backwards compatible or is it a breaking change?
backwards compatible

### CODE REVIEW

**Comment by jcfilben** [comment]
Marking as a draft. I realized the close icon button alignment is working well for the hpe theme but not for the grommet theme. Going back to see if I can find a solution that works for both

**Comment by jcfilben** [comment]
I tested out different options for the close button icon alignment and found that having a top pad of 'small' works better for the grommet theme and 'xsmall' works better for the hpe theme (see line 353 of SelectMultipleContainer.js).

For now I left the padding as 'small' as it felt visually slightly better than setting padding to 'xsmall'

File: src/js/components/SelectMultiple/SelectMultiple.js [hunk 1]
```diff
@@ -421,10 +421,12 @@ const SelectMultiple = forwardRef(
...
                         tabIndex="-1"
                         type="text"
-                        // eslint-disable-next-line max-len
                         placeholder={
+                          // eslint-disable-next-line no-nested-ternary
                           value.length === 0
                             ? placeholder || selectValue || displayLabelKey
+                            : onMore
+                            ? `${value.length} selected`
                             : `${value.length} selected of ${allOptions.length}`
```
Comment (ericsoderberghp) [inline src/js/components/SelectMultiple/SelectMultiple.js:430]: OK for now, but these should be included in i18n and use MessageContext at some point. Can you file a separate issue for that?

**Review by ericsoderberghp — APPROVED** [review]

File: src/js/components/SelectMultiple/SelectMultiple.js (hunk 1 above)
Comment (jcfilben) [reply src/js/components/SelectMultiple/SelectMultiple.js:430]: Yes, I'll create a separate issue

Comment (jcfilben) [reply src/js/components/SelectMultiple/SelectMultiple.js:430]: Issue created https://github.com/grommet/grommet/issues/6337

**Comment by taysea** [comment]
Maybe an enhancement separate, but wondering if we should allow the caller to control `totalOptions` (num) and `onSelectAll` (func) to handle large data cases?

**Comment by ericsoderberghp** [comment]
> Maybe an enhancement separate, but wondering if we should allow the caller to control `totalOptions` (num) and `onSelectAll` (func) to handle large data cases?

Good thoughts. I am willing to move forward without for now.

**Review by taysea — APPROVED** [review]

**Pull request merged**